### PR TITLE
[wasm] Docs on testing blazor wasm

### LINKED ISSUE #85413: Can we have better doc on inner loop workflow for testing runtime changes in a BlazorWasm app?
I found the process to test these with a BlazorWasm app to be somewhat cumbersome and error-prone. I didn't find and doc on this, such as at https://github.com/dotnet/runtime/blob/main/src/mono/wasm/build/README.md. 

I do not make changes specifically for BlazorWasm or Mono interpreter often, so here's my primitive and brute-force workflow for getting library and runtime changes so I can test those within a Blazor app, which I'm sure is not ideal:
1) Make runtime and\or library changes (e.g. in `C:\git\runtime`)
2) Copy those changes to the current package location. If copied locally into the WASM app location bin staging area, these seem to get overridden during publish.
    - e.g. for runtime changes: `copy C:\git\runtime\artifacts\bin\microsoft.netcore.app.runtime.browser-wasm\Release\runtimes\browser-wasm\native\System.Private.CoreLib.dll C:\Users\<YOURNAME>\.nuget\packages\microsoft.netcore.app.runtime.mono.browser-wasm\8.0.0-preview.3.23174.8\runtimes\browser-wasm\native\`.
    - This is super fragile and assumes my local build is compatible with the referenced version in the app's .csproj's `PackageReference`.
    - Some alternatives, partially explored, is to either add a `<Reference Include>` to the .csproj to point to a local copy of `System.Private.CoreLib.dll`, for example, which would override the package version, or publish my local build and do some nuget package redirection. Not sure what devs that work in this area do.
3) In the BlazorWasm app folder, delete the build artifacts so that publish will update the staging area: `del /q /s obj\Release\net8.0 && del /q /s bin\Release\net8.0`
4) Publish: `dotnet publish -c release`
5) Change directory to `wwwroot` (it appears the active directory must be `wwwroot`): `cd bin\Release\net8.0\publish\wwwroot\`
6) Run: `dotnet serve`

### PR DESCRIPTION
**Pull request by maraf**
- Add `UpdateRuntimePack` hint

Closes https://github.com/dotnet/runtime/issues/85413

### CODE REVIEW

File: src/mono/wasm/build/README.md [hunk 1]
```diff
@@ -140,4 +140,17 @@ If encountering build performance issues, you can use the rollup `--perf` option
...
+
+## Blazor
+
+You can test local build of runtime in blazor by overriding location where blazor resolves runtime pack from
+
+*BlazorWasm.csproj*
+```xml
+<Target Name="UpdateRuntimePack" AfterTargets="ResolveFrameworkReferences">
+  <ItemGroup>
+    <ResolvedRuntimePack PackageDirectory="{RUNTIME_REPO_ROOT}\artifacts\bin\microsoft.netcore.app.runtime.browser-wasm\Debug" Condition="'%(ResolvedRuntimePack.FrameworkName)' == 'Microsoft.NETCore.App'" />
```
Comment (lambdageek) [inline src/mono/wasm/build/README.md:153]: for testing AOT do you also need to point at a different Cross AOT package?

Comment (lambdageek) [reply src/mono/wasm/build/README.md:153]: For an ItemGroup don't you usually need `Include=` or `Update=` ? what does it do if you don't one of those?

Comment (maraf) [reply src/mono/wasm/build/README.md:153]: Update works only if items are defined outside of a target. This updates all items matching the condition

Comment (steveisok) [reply src/mono/wasm/build/README.md:153]: > for testing AOT do you also need to point at a different Cross AOT package?

Yes - I don't know if this is too early, but you could clear the `MonoAotCrossCompiler` item and provide a new one that points in tree.

Comment (maraf) [reply src/mono/wasm/build/README.md:153]: I have never done that, you have a snippet for it?

**Review by lewing — APPROVED** [review]

**Pull request merged**